src/test/scala/JointSpec.scala

```diff
@@ -20,65 +19,175 @@ class JointSpec extends FunSuite {
+                      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == math.Pi/2.0)
+  }
+
+  test("Dip direction for vertical joint should be pi/2.0") {
+    val joint = Joint((0.0, -1.0, 0.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == math.Pi/2.0)
+  }
+
+  test("Dip direction for vertical joint should be pi") {
+    val joint = Joint((-1.0, 0.0, 0.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == math.Pi)
+  }
+
+  test("Dip direction for vertical joint should be 3.0*pi/2.0") {
+    val joint = Joint((0.0, 1.0, 0.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == 3.0*2.0*math.Pi/4.0)
+  }
+
+  test("Dip direction for non-vertical joint should be pi/2.0") {
+    val joint = Joint((0.0,-1.0/math.sqrt(2.0), 1.0/math.sqrt(2.0)), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == math.Pi/2.0)
+  }
+
+  test("Dip direction for non-vertical joint should be 3.0*pi/4.0") {
+    val joint = Joint((-1.0/math.sqrt(3.0),-1.0/math.sqrt(3.0), 1.0/math.sqrt(3.0)), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == 3.0*math.Pi/4.0)
+  }
+
+  test("Dip direction for non-vertical joint should be 7.0*pi/4.0") {
+    val joint = Joint((1.0/math.sqrt(3.0), 1.0/math.sqrt(3.0), 1.0/math.sqrt(3.0)), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == 7.0*math.Pi/4.0)
+  }
+
+  test("Dip angle for horizontal joint should be 0.0") {
+    val joint = Joint((0.0, 0.0, 1.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipAngle == 0.0)
+  }
+
+  test("Dip angle for vertical joint should be pi/2.0") {
+    val joint = Joint((1.0, 0.0, 0.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipAngle == math.Pi/2.0)
+  }
+
+  test("Dip angle for joint with positive z-component in normal should be pi/6.0") {
+    val joint = Joint((1.0/math.sqrt(1.0 + math.sqrt(3.0)), 0.0, math.sqrt(3.0)/math.sqrt(1.0 + math.sqrt(3.0))), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(math.abs(joint.dipAngle - math.Pi/6.0) < EPSILON)
+  }
+
+  test("Dip angle for joint with negative z-component in normal should be pi/6.0") {
+    val joint = Joint((0.0, 1.0/math.sqrt(1.0 + math.sqrt(3.0)), -math.sqrt(3.0)/math.sqrt(1.0 + math.sqrt(3.0))), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(math.abs(joint.dipAngle - math.Pi/6.0) < EPSILON)
+  }
+
+  test("Dip direction of pi/2.0 centered at origin with positive z component in normal") {
```

Comment:
From what I've seen, the convention for these test description strings is to state the desired outcome of the test too, so that it's easier to see what's going on when you do something like an `sbt test`. I know this is kind of picky, but would you mind adding that in where it's missing?

For better or for worse, part of the whole software engineering thing is sticking to consistent conventions. It's a pain and a lot of CS people don't do this as well as they should, but we want to be a good example right? :)